File: app/src/main/java/com/andysklyarov/finnotify/presentation/NamesListDialog.java
package com.andysklyarov.finnotify.presentation;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDialogFragment;

import com.andysklyarov.finnotify.R;

public class NamesListDialog extends AppCompatDialogFragment {

    private NamesListDialogListener listener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        try {
            listener = (NamesListDialogListener) getTargetFragment();
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() +
                    "Must implement NamesListDialogListener");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        Activity activity = getActivity();
        AlertDialog.Builder builder = new AlertDialog.Builder(activity, R.style.AlertDialogCustom);

        LayoutInflater inflater = activity.getLayoutInflater();
        View view = inflater.inflate(R.layout.dialog_layout, null);

        builder.setView(view)
                .setTitle(R.string.dialog_title)
                .setItems(R.array.currencies, (dialogInterface, i) -> {
                    String code = getResources().getStringArray(R.array.currencies)[i];
                    listener.applyCode(getTargetRequestCode(), Activity.RESULT_OK, code);
                });

        AlertDialog dialog = builder.create();
        dialog.getWindow().setDimAmount(0.85f);
        dialog.getWindow().setBackgroundDrawableResource(android.R.color.transparent);
        return dialog;
    }

    public interface NamesListDialogListener {
        void applyCode(int requestCode, int resultCode, String NameAndCode);
    }
}

File: app/src/main/java/com/andysklyarov/finnotify/presentation/NamesListDialog.kt
package com.andysklyarov.finnotify.presentation

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatDialogFragment
import com.andysklyarov.finnotify.R


class NamesListDialog : AppCompatDialogFragment() {

    private lateinit var listener: NamesListDialogListener


    override fun onAttach(context: Context) {
        super.onAttach(context)

        listener = try {
            (targetFragment as NamesListDialogListener)
        } catch (e: ClassCastException) {
            throw ClassCastException(
                context.toString().toString() +
                        "Must implement NamesListDialogListener"
            )
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity, R.style.AlertDialogCustom)

        val inflater = activity?.layoutInflater
        val view = inflater?.inflate(R.layout.names_dialog_layout, null)

        builder.setView(view)
            .setTitle(R.string.names_dialog_title)
            .setItems(R.array.currencies){ _, i ->
                val code = resources.getStringArray(R.array.currencies)[i]
                listener.applyCode(targetRequestCode, Activity.RESULT_OK, code)
            }

        val dialog = builder.create()
        dialog.window?.setDimAmount(resources.getFloat(R.dimen.dialog_dim_amount))
        dialog.window?.setBackgroundDrawableResource(R.color.dialog_background)

        return dialog
    }

    interface NamesListDialogListener {
        fun applyCode(requestCode: Int, resultCode: Int, NameAndCode: String)
    }
}